test/unittest/unittest_nntrainer_internal.cpp

```diff
@@ -773,20 +788,171 @@ TEST(nntrainer_DataBuffer, init_02_n) {
+      }
+    }
+  }
+
+  Results = T.apply(nntrainer::softmax);
+  float *data = Results.getData();
+
+  for (int i = 0; i < batch * height * width; ++i) {
+    if ((data[i] - results[i % width]) > tolerance) {
+      status = ML_ERROR_INVALID_PARAMETER;
+      break;
+    }
+  }
+  EXPECT_EQ(status, ML_ERROR_NONE);
+}
+
+TEST(nntrainer_util_func, random_01_p) {
+  int status = ML_ERROR_INVALID_PARAMETER;
+  srand(time(NULL));
+  float x = nntrainer::random(0.0);
+  if (-1.0 < x && x < 1.0)
+    status = ML_ERROR_NONE;
+  EXPECT_EQ(status, ML_ERROR_NONE);
+}
+
+TEST(nntrainer_util_func, sqrtFloat_01_p) {
+  int status = ML_ERROR_INVALID_PARAMETER;
+
+  float x = 9871.0;
+  float sx = nntrainer::sqrtFloat(x);
+
+  if ((sx * sx - x) * (sx * sx - x) < tolerance)
+    status = ML_ERROR_NONE;
+  EXPECT_EQ(status, ML_ERROR_NONE);
+}
+
+TEST(nntrainer_util_func, logFloat_01_p) {
+  int status = ML_ERROR_NONE;
+  int batch = 1;
+  int height = 1;
+  int width = 10;
+
+  nntrainer::Tensor input(batch, height, width);
+  GEN_TEST_INPUT(i * (width) + k + 1);
+
+  nntrainer::Tensor Results = input.apply(nntrainer::logFloat);
+
+  float *data = Results.getData();
+  float *indata = input.getData();
+
+  for (int i = 0; i < batch * height * width; ++i) {
+    if ((data[i] - (float)log(indata[i])) > tolerance) {
+      status = ML_ERROR_INVALID_PARAMETER;
+      break;
+    }
+  }
+
+  EXPECT_EQ(status, ML_ERROR_NONE);
+}
+
+TEST(nntrainer_util_func, sigmoid_01_p) {
+  int status = ML_ERROR_NONE;
+  int batch = 1;
+  int height = 1;
+  int width = 10;
+
+  nntrainer::Tensor input(batch, height, width);
+  GEN_TEST_INPUT((k - 4) * 0.1);
+
+  nntrainer::Tensor Results = input.apply(nntrainer::sigmoid);
+
+  float *data = Results.getData();
+  float *indata = input.getData();
+
+  for (int i = 0; i < batch * height * width; ++i) {
+    if ((data[i] - (float)(1 / (1 + exp(-indata[i])))) > tolerance) {
+      status = ML_ERROR_INVALID_PARAMETER;
+      break;
+    }
+  }
+
+  EXPECT_EQ(status, ML_ERROR_NONE);
+}
+
+TEST(nntrainer_util_func, tanhFloat_01_p) {
+  int status = ML_ERROR_NONE;
+  int batch = 1;
+  int height = 1;
+  int width = 10;
+
+  nntrainer::Tensor input(batch, height, width);
+  GEN_TEST_INPUT((k - 4) * 0.1);
+
+  nntrainer::Tensor Results = input.apply(nntrainer::tanhFloat);
+
+  float *data = Results.getData();
+  float *indata = input.getData();
+
+  for (int i = 0; i < batch * height * width; ++i) {
+    if ((data[i] - (float)(tanh(indata[i]))) > tolerance) {
+      status = ML_ERROR_INVALID_PARAMETER;
+      break;
+    }
+  }
+
+  EXPECT_EQ(status, ML_ERROR_NONE);
+}
+
+TEST(nntrainer_util_func, relu_01_p) {
+  int status = ML_ERROR_NONE;
+  int batch = 1;
+  int height = 1;
+  int width = 10;
+
+  nntrainer::Tensor input(batch, height, width);
+  GEN_TEST_INPUT((k - 4) * 0.1);
+
+  nntrainer::Tensor Results = input.apply(nntrainer::relu);
+
+  float *data = Results.getData();
+  float *indata = input.getData();
 
```

Comment:
How about adding ASSERT_NE(nullptr, data) and ASSERT_NE(nullptr, indata)?

Reply:
added! Thanks.